### PR TITLE
[V18: Volume Split] Refactor create gauge to only interface with pool ID

### PR DESCRIPTION
**Pull request by stackman27**
Part of: #[6057](https://github.com/osmosis-labs/osmosis/issues/6057)

## What is the purpose of the change
indepth overview of this change is in this hackmd (https://hackmd.io/rLt-8Ox4Qn2pIXRnsLyhMg) 
but here's a tldr 
- [x] Remove distrTo while creating gauges (not needed for CL, and GAMM only uses 14d incentives so we can have 1:1 mapping with gauge and PoolId rn) 
- [x] After Pool Create Epoch, create gauge with longestLockableDuration (14d only) (this doesnot mean we remove 1d, 7d from lockableDurations on chain) 
- [x] Fix all broken test 
- [x] Add new test for CreateGauge

## Testing and Verifying

-localosmosis tested 

```
**** FOR CL *****
- osmosisd tx incentives create-gauge "" 100000uosmo 3 --from val --keyring-backend test --chain-id localosmosis  --epochs 12 --fees 100000uosmo -b block -y

- osmosisd q poolincentives external-incentivized-gauges                                                                                                    
data:
- coins:
  - amount: "100000"
    denom: uosmo
  distribute_to:
    denom: no-lock/e/3
    duration: 0s
    lock_query_type: NoLock
    timestamp: "0001-01-01T00:00:00Z"
  distributed_coins:
  - amount: "8333"
    denom: uosmo
  filled_epochs: "1"
  id: "5"
  is_perpetual: false
  num_epochs_paid_over: "12"
  start_time: "1970-01-01T00:00:00Z"

***** FOR NON CL (GAMM) ****
-osmosisd tx incentives create-gauge gamm/pool/1 100000uosmo 1 --from val --keyring-backend test --chain-id localosmosis  --epochs 12 --fees 100000uosmo -b block -y

-osmosisd q poolincentives external-incentivized-gauges                                                                                                             
data:
- coins:
  - amount: "100000"
    denom: uosmo
  distribute_to:
    denom: gamm/pool/1
    duration: 240s
    lock_query_type: ByDuration
    timestamp: "0001-01-01T00:00:00Z"
  distributed_coins: []
  filled_epochs: "0"
  id: "6"
  is_perpetual: false
  num_epochs_paid_over: "12"
  start_time: "1970-01-01T00:00:00Z"
- coins:
  - amount: "100000"
    denom: uosmo
  distribute_to:
    denom: no-lock/e/3
    duration: 0s
    lock_query_type: NoLock
    timestamp: "0001-01-01T00:00:00Z"
  distributed_coins:
  - amount: "16666"
    denom: uosmo
  filled_epochs: "2"
  id: "5"
  is_perpetual: false
  num_epochs_paid_over: "12"
  start_time: "1970-01-01T00:00:00Z"



``` 

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/gamm/keeper/migrate_test.go [hunk 1]
```diff
@@ -1132,13 +1132,13 @@ func (s *KeeperTestSuite) TestCreateCanonicalConcentratedLiquidityPoolAndMigrati
 			desiredDenom0:        USDCIBCDenom,
 			expectError:          types.ErrMustHaveTwoDenoms,
 		},
-		"error: invalid denom durations": {
```
Comment (stackman27) [inline x/gamm/keeper/migrate_test.go:1135]: note: cannot have this scenerio, lockableDuration must be set at genesis or onChain and we fetch longestDuration afterCreateBalancerPool hook

File: x/incentives/keeper/gauge_test.go [hunk 2]
```diff
@@ -14,41 +14,39 @@ import (
 
 var _ = suite.TestingSuite(nil)
 
+// NOTE: this is no-op because we internally calculate duration
 // TestInvalidDurationGaugeCreationValidation tests error handling for creating a gauge with an invalid duration.
```
Comment (stackman27) [inline x/incentives/keeper/gauge_test.go:18]: // NOTE: this is no-op because we internally calculate duration

**Comment by stackman27** [comment]
@devbot-wizard help

**Comment by devbot-wizard** [comment]
Hi! I'm DevBot, a bot that helps with common tasks in the development process. Commands:
- `devbot fix`: Fix basic errors in the PR. (e.g. imports, changelog conflicts, go mod conflicts)
- `devbot merge base`: Merge the base branch into the PR branch.
- `devbot add changelog [misc/feat/bug/api/sec] [message]`: Add a changelog entry to the PR. (e.g. `devbot add changelog feat Added a new feature`)
    - If message is blank, defaults to PR title.
- `devbot re-pr`: Re-PR the PR, useful for external contributors where we have no edit perms.

**Comment by stackman27** [comment]
devbot add changelog v18: Volume Split, Refactor create gauge to only interface with pool ID

File: x/incentives/keeper/gauge.go [hunk 3]
```diff
@@ -173,6 +171,9 @@ func (k Keeper) CreateGauge(ctx sdk.Context, isPerpetual bool, owner sdk.AccAddr
 		if !k.bk.HasSupply(ctx, distrTo.Denom) && !strings.Contains(distrTo.Denom, "osmovaloper") {
 			return 0, fmt.Errorf("denom does not exist: %s", distrTo.Denom)
 		}
+
+		// for all external incentives by duration create general linking as well
+		k.pik.SetGeneralGaugeToPoolLink(ctx, nextGaugeId, longestDuration, poolId)
```
Comment (stackman27) [inline x/incentives/keeper/gauge.go:176]: we added 1:1 mapping for gaugeId <> poolId for duration based external incentives which doesnot exist rn. So we might have to run migrations to accomodate existing gauges

**Pull request closed without merging**